servicetest/coredump/test_coredump.py

```diff
@@ -0,0 +1,175 @@
+
+# These default values are used to pass various test specific values and
+# configurations to the Container helper. Tests that need to add, remove or
+# update entries can simply base their dict on this one for convenience.
+DATA = {
+    Container.CONFIG: '[{"enableWriteBuffer": false}]',
+    Container.BIND_MOUNT_DIR: "/gateways/app",
+    Container.HOST_PATH: CURRENT_DIR,
+    Container.READONLY: False
+}
+
+# This function is used by the testframework 'testhelper' fixture to know where the
+# testhelper should be made available when running the tests
+def mounted_path_in_host():
+    return CURRENT_DIR
+
+#
+# Gateway config and capability + service manifest for enabling core dumps
+#
+FILE_CONFIG = [{
+    "path-host": coredump_path(),
+    "path-container": coredump_path(),
+    "read-only": False
+}]
+default_cap = Capability("test.cap.default.coredump",
+                        [
+                            { "id": "file", "config": FILE_CONFIG }
+                        ])
+default_manifest = DefaultManifest(TESTOUTPUT_DIR, "default-caps.json", [default_cap])
+
+def service_manifests():
+    """ The agent fixture calls this function when it creates the service manifests
+        that should be used with this test module. The agent fixture expects a list
+        of StandardManifest and/or DefaultManifest objects.
+    """
+    return [default_manifest]
+
+@pytest.fixture
+def create_testoutput_dir(scope="module"):
+    """ Create a directory for the generated test files.
+
+        This directory is ignored by git but it's nice to have
+        somewhere locally to store test output to support
+        troubleshooting etc.
+    """
+    if not os.path.exists(TESTOUTPUT_DIR):
+        os.makedirs(TESTOUTPUT_DIR)
+
+@pytest.fixture
+def create_coredump_dir(scope="module"):
+    """ Create the directories needed for getting the core dumps
+    """
+
+    # Create the core dump directory if it doesn't already exist
+    if not os.path.exists(coredump_path()):
+        os.makedirs(coredump_path())
+
+    yield
+
+    shutil.rmtree(coredump_path())
+    assert(not os.path.exists(coredump_path()))
+
+@pytest.fixture
+def set_core_pattern(scope="function"):
+    """ Check the core pattern and set it to something usable for this test
+
+        Also reset the pattern at test end
+    """
+
+    pattern_path = "/proc/sys/kernel/core_pattern"
+    test_pattern = coredump_path() + "/coredump"
+    old_pattern = ""
+
+    with open(pattern_path, "r") as core_file:
+        old_pattern = core_file.read().strip()
+
+    with open(pattern_path, "w") as core_file:
+        core_file.write(test_pattern)
+
+    with open(pattern_path, "r") as core_file:
+        assert(core_file.read().strip() == test_pattern)
+
+    yield test_pattern
+
+    # Teardown
+    with open(pattern_path, "w") as core_file:
+        core_file.write(old_pattern)
+
+    with open(pattern_path, "r") as core_file:
+        assert(core_file.read().strip() == old_pattern)
```

Comment:
I think we could remove these asserts that are not part of the test. Unless there's a good reason for us to not trust that the file operations went as expected.